pkg/template_generator.go

```diff
@@ -130,6 +130,19 @@ func generateTemplateForPodSpec(podSpec kapi.PodSpec, key string, value map[stri
 		podSpec.RestartPolicy = kapi.RestartPolicy(fmt.Sprintf("{{.Values.%s.%s}}", key, RestartPolicy))
 	}
 
+	if len(podSpec.ImagePullSecrets) != 0 {
+		imagePullSecretsObj := []kapi.LocalObjectReference{}
+
+		for _, imagePullSecrets := range podSpec.ImagePullSecrets {
+			imagePullSecrets.Name = generateSafeKey(imagePullSecrets.Name)
+			value["imagePullSecrets"] = imagePullSecrets.Name
```

Comment:
As i see, if i put a secret.yml in input, the name is changed by using generateSafeKey(). So, obviously the deployment will not get correct secret name while deploying to K8s. Appropriate name must be set in deployment.yml as well. So, i made it configurable & modified name the same way secret name was modified from secret.yml.

I think it should be directly mapped to secret name from secret.yml by some way. 
Any suggestions what can be done here?